tools/s3_to_seq/s3_to_seq.py

```diff
@@ -0,0 +1,328 @@
+
+        Attributes:
+            url (str): The URL of the Seq server
+            batch (bytes): A batch of log messages waiting to be sent
+            ctr (int): A counter for the number of log messages sent
+        """
+        self.url = url
+        self.batch = b""
+        self.ctr = 0
+
+    def __enter__(self):
+        """
+        Enter a context and return the LogSender instance
+
+        Returns:
+            LogSender: Instance that support the with statement
+        """
+        return self
+
+    def sendlog(self, line: str):
+        """
+        Send a log message to the Seq server. The message is expected to be in JSON format
+        Logs are sent to the server when the batch size exceeds 100,000 bytes
+
+        Args:
+            line (str): A log message in JSON format
+        """
+        ctx = Context()
+        if line.startswith("{"):
+            try:
+                parsed = json.loads(line)
+                if "@t" not in parsed:
+                    return
+                self.ctr = self.ctr + 1
+                log_message = bytes(line + "\n", "utf-8")
+                if len(self.batch) + len(log_message) > 100000:
+                    requests.post(self.url, data=self.batch)
+                    self.batch = log_message
+                else:
+                    self.batch += log_message
+            except json.JSONDecodeError as e:
+                ctx.logger.warning(f"Failed to parse JSON: {e}")
+
+    def __exit__(self, exception_type, exception_value, traceback):
+        """
+        Logs left in the batch are sent and exiting the context
+        """
+        ctx = Context()
+        if self.batch != b"":
+            requests.post(self.url, data=self.batch)
+        ctx.logger.debug(f"Remaining {self.ctr} elements sent, exiting a LogSender context")
+
+
+def process_json_log(url: str, file_name: str):
+    """
+    Process a JSON log file and send its contents to a Seq server
+
+    Args:
+        url (str): The URL of the Seq server where log messages should be sent
+        file_name (str): The path to the JSON log file
+    """
+    ctx = Context()
+    ctx.logger.debug("Sending json file : " + file_name)
+    with open(file_name, "r") as file:
+        with LogSender(url) as log_sender:
+            for line in file.readlines():
+                log_sender.sendlog(line)
+
+
+def process_jsongz_log(url: str, file_name: str):
+    """
+    Process a gzipped JSON log file and send its contents to a Seq server
+
+    Args:
+        url (str): The URL of the Seq server where log messages should be sent
+        file_name (str): The path to the gzipped JSON log file
+    """
+    ctx = Context()
+    ctx.logger.debug("Sending gzipped json file : " + file_name)
+    with gzip.open(file_name, "r") as file:
+        with LogSender(url) as log_sender:
+            for line in file.read().decode("utf-8").split("\n"):
+                log_sender.sendlog(line)
+
+
+def main(args: argparse.Namespace):
```

Comment:
Do not use argument namespace as function input. Instead list arguments and then call the function using **args to unpack the namespace.